x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel_wrapper.tsx

```diff
@@ -138,6 +144,9 @@ export function WorkspacePanelWrapper({
   const maxDimensions = displayOptions?.maxDimensions;
   const minDimensions = displayOptions?.minDimensions;
 
+  const [resizeRef, setResizeRef] = React.useState<Element | null>(null);
+  const { width, height } = useResizeObserver(resizeRef, 'width');
+
   let visDimensionsCSS: Interpolation<Theme> = {};
 
   if (aspectRatio) {
```

Comment:
In this implementation the `maxDimensions` clamping is still needed. At the boundary of the constraint 

```ts
  if (maxDimensions) {
    visDimensionsCSS.maxWidth = maxDimensions.x
      ? `min(100%, ${maxDimensions.x.value}${unitToCSSUnit[maxDimensions.x.unit]})`
      : '';
    visDimensionsCSS.maxHeight = maxDimensions.y
      ? `min(100%, ${maxDimensions.y.value}${unitToCSSUnit[maxDimensions.y.unit]})`
      : '';
  }
```

![Zight Recording 2024-10-16 at 12 14 09 PM](https://github.com/user-attachments/assets/57efb2d6-4ee8-4778-9c76-b8c47db86f83)